test/DynamoCoreWpfTests/NodeAutoCompleteSearchTests.cs

```diff
@@ -0,0 +1,121 @@
+        {
+            libraries.Add("FunctionObject.ds");
+            libraries.Add("BuiltIn.ds");
+            libraries.Add("FFITarget.dll");
+        }
+
+        public override void Open(string path)
+        {
+            base.Open(path);
+
+            DispatcherUtil.DoEvents();
+        }
+
+        public override void Run()
+        {
+            base.Run();
+
+            DispatcherUtil.DoEvents();
+        }
+
+
+        [Test]
+        public void NodeSuggestions_InputPortZeroTouchNode_AreCorrect()
+        {
+            Open(@"UI\ffitarget_inputport_suggestion.dyn");
+
+            // Get the node view for a specific node in the graph
+            NodeView nodeView = NodeViewWithGuid(Guid.Parse("9aeba33453a34c73823976222b44375b").ToString());
+
+            var inPorts = nodeView.ViewModel.InPorts;
+            Assert.AreEqual(2, inPorts.Count());
+
+            var port = inPorts[0].PortModel;
+            var type = port.GetInputPortType();
+            Assert.AreEqual("FFITarget.DummyPoint", type);
+
+            port = inPorts[1].PortModel;
+            type = port.GetInputPortType();
+            Assert.AreEqual("FFITarget.DummyVector", type);
+
+            var searchViewModel = (ViewModel.CurrentSpaceViewModel.NodeAutoCompleteSearchViewModel as NodeAutoCompleteSearchViewModel);
+            searchViewModel.PortViewModel = inPorts[1];
+            var suggestions = searchViewModel.GetMatchingNodes();
+            Assert.AreEqual(5, suggestions.Count());
+
+            var suggestedNodes = suggestions.Select(s => s.FullName).OrderBy(s => s);
+            var nodes = new[] { "FFITarget.FFITarget.DummyVector.DummyVector", "FFITarget.FFITarget.DummyVector.ByCoordinates",
+                "FFITarget.FFITarget.DummyVector.ByVector", "FFITarget.FFITarget.DummyVector.Scale", "FFITarget.FFITarget.DummyPoint.DirectionTo" };
+            var expectedNodes = nodes.OrderBy(s => s);
+            for (int i = 0; i < 5; i++)
+            {
+                Assert.AreEqual(expectedNodes.ElementAt(i), suggestedNodes.ElementAt(i));
+            }
+        }
+
+        [Test]
+        public void NodeSuggestions_InputPortBuiltInNode_AreCorrect()
+        {
+            Open(@"UI\builtin_inputport_suggestion.dyn");
+
+            // Get the node view for a specific node in the graph
+            NodeView nodeView = NodeViewWithGuid(Guid.Parse("b6cb6ceb21df4c7fb6b186e6ff399afc").ToString());
+
+            var inPorts = nodeView.ViewModel.InPorts;
+            Assert.AreEqual(2, inPorts.Count());
+
+            var port = inPorts[0].PortModel;
+            var type = port.GetInputPortType();
+            Assert.AreEqual("var[]..[]", type);
+
+            port = inPorts[1].PortModel;
+            type = port.GetInputPortType();
+            Assert.AreEqual("string", type);
+
+            var searchViewModel = (ViewModel.CurrentSpaceViewModel.NodeAutoCompleteSearchViewModel as NodeAutoCompleteSearchViewModel);
```

Comment:
I was hoping we could instantiate a new `NodeAutoCompleteSearchViewModel` with a new `PortViewModel` each time we wanted autocomplete, something like:
```
var searchViewModel = new NodeAutoCompleteSearchViewModel(inPorts[1]);
var suggestions = searchViewModel.GetMatchingNodes();
```
as I thought from a usability standpoint of the API this looks cleaner but unfortunately, we can't do that as it is currently designed I suppose.

Reply:
hmm since the base class of `NodeAutoCompleteSearchViewModel` take DynamoViewModel from constructor, it is currently not doable but we could write an alternative constructor takes both. Well, I guess I am curious if you see other benefits of initializing it more than doing it only once?

Reply:
The only benefit I see is from a code clarity standpoint. It is a trade-off between:
1. Use a single instance of `NodeAutoCompleteSearchViewModel` and replace its contents each time by accessing the same object each time from the workspace model vs
2. Create a new temporary object each time, use it for autocomplete where needed and forget about it

Reply:
I agree. Since we adopted a majority code of InCanvasSearchViewModel which is a singleton, it may make sense to touch how this works later? If it is not urgent.. What do you think?